### PR TITLE
Refactor getVisitSlotsFromBookedVisits and getPrisonerEvents

### PR DESCRIPTION
**Pull request by psoleckimoj**

### CODE REVIEW

**Comment by psoleckimoj** [comment]
Moved a function to allow for later testing, also refactored function to make extending easier and to make logic cleaner.

File: server/utils/visitsUtils.ts [hunk 1]
```diff
@@ -1,67 +1,71 @@
...
-        })
-
-        matchingOpenSlot = openSlots.length - 1
-      }
-
-      openSlots[matchingOpenSlot].adults += visit.visitors.filter(visitor => visitor.adult).length
-      openSlots[matchingOpenSlot].children += visit.visitors.filter(visitor => !visitor.adult).length
-    } else {
-      let matchingClosedSlot = closedSlots.findIndex(closedSlot => closedSlot.visitTime === visit.visitTime)
+    const visitRestriction = visit.visitRestriction === 'OPEN' ? 'OPEN' : 'CLOSED'
```
Comment (psoleckimoj) [inline server/utils/visitsUtils.ts:22]: Currently dumps UNKNOWN in with CLOSED

Comment (psoleckimoj) [reply server/utils/visitsUtils.ts:22]: Awaiting design updates on this to do it properly

**Review by hutcheonb-moj — APPROVED** [review]
LGTM

**Pull request merged**